### PR TITLE
Add a CONTRIBUTING file

### LINKED ISSUE #425: let's add a CONTRIBUTING.md file
as pointed by Pascal https://twitter.com/pascalchevrel/status/695533836113936384

Maybe we can have it there and somehow pull it in Sphinx ?

### PR DESCRIPTION
**Pull request by magopian**
Fixes #425

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.. _contributing:
+
+============
+Contributing
+============
+
+Before contributing code, please note:
+
+* You agree to license your contributions under the [license](https://github.com/mozilla/kinto/blob/master/LICENSE).
+* Please read the [contributing documentation](http://kinto.readthedocs.org/en/latest/contributing.html).
```
Comment (almet) [inline CONTRIBUTING.rst]: That's two times "please" :)

**Comment by almet** [comment]
General note: does it work with a `.rst` extension?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.. _contributing:
+
+============
+Contributing
+============
```
Comment (almet) [inline CONTRIBUTING.rst]: I believe we don't need to add a specific header here.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.. _contributing:
```
Comment (almet) [inline CONTRIBUTING.rst]: nor a link :)

File: CONTRIBUTING.rst (hunk 1 above)
Comment (magopian) [inline CONTRIBUTING.rst]: I'm extra polite ;)

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+============
+Contributing
+============
+
+Before contributing code, please note:
```
Comment (almet) [inline CONTRIBUTING.rst]: That could be neat to add a short sentence saying "Thanks for considering submitting a pull request, we really appreciate that. Before doing so, here are a few guidelines"

**Comment by magopian** [comment]
It does work with a .rst extension, I double checked, but it could also be a .md file if you'd rather have that, what do you think?

File: CONTRIBUTING.rst (hunk 1 above)
Comment (almet) [inline CONTRIBUTING.rst]: We could quickly extract the most important things from this doc, put them here and add a link to "see you to contribute in more details"?

Comment (almet) [inline CONTRIBUTING.rst]: We could quickly extract the most important things from this doc, put them here and add a link to "see you to contribute in more details"?

**Comment by almet** [comment]
Either is fine! (a.k.a "I don't care")

**Comment by magopian** [comment]
@almet I've updated the file, let me know what you think ;)

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
+Thanks for considering submitting a pull request, we really appreciate that.
```
Comment (almet) [inline CONTRIBUTING.rst]: nit: an exclamation mark could add some more energy in the phrasing?

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+Before doing so, here are a few guidelines:
+
+* You agree to license your contributions under the
+  [license](https://github.com/mozilla/kinto/blob/master/LICENSE).
+* Report bugs as [new issues on
```
Comment (almet) [inline CONTRIBUTING.rst]: I believe we don't really need to point that out.

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  github](https://github.com/Kinto/kinto/issues/new).
+* Use pull-requests early so it's open for discussion, even if your
+  contribution isn't ready yet.
+* A pull-request should include tests.
```
Comment (almet) [inline CONTRIBUTING.rst]: All pull requests should include tests, they help us avoiding regressions in our code.

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+* A pull-request adding functionality should also include docs.
+* [TravisCI integration tests](https://travis-ci.org/Kinto/kinto) should be
+  green.
+* Check the [contributing
```
Comment (almet) [inline CONTRIBUTING.rst]: If you're interested, you can also [...]

**Comment by almet** [comment]
LGTM with nits.

**Comment by magopian** [comment]
Thanks a lot for the thorough review @almet ! I've updated it, let me know if that still looks good to you

File: CONTRIBUTING.rst [hunk 9]
```diff
@@ -0,0 +1,15 @@
+Thanks for considering submitting a pull request, we really appreciate that!
+Before doing so, here are a few guidelines:
```
Comment (almet) [inline CONTRIBUTING.rst]: nit: Add a line break between the two lines?

File: CONTRIBUTING.rst [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+* You agree to license your contributions under the
+  [license](https://github.com/mozilla/kinto/blob/master/LICENSE).
+* Use pull-requests early so it's open for discussion, even if your
+  contribution isn't ready yet.
+* All pull requests should include tests, they help us avoiding regressions in
+  our code.
+* A pull-request adding functionality should also include docs.
```
Comment (almet) [inline CONTRIBUTING.rst]: should also update the documentation accordingly

**Comment by almet** [comment]
Just noted that links do us the `markdown` format whereas this is `rst`.

File: CONTRIBUTING.rst (hunk 10 above)
Comment (magopian) [inline CONTRIBUTING.rst]: What do you mean? I took this line from http://kinto.readthedocs.org/en/latest/contributing.html#pull-request-guidelines

Comment (almet) [inline CONTRIBUTING.rst]: It's just another way to phrase it (docs vs documentation)

Comment (magopian) [inline CONTRIBUTING.rst]: ah, sorry, I thought you meant I should update the contributing.html documentation file ;) Thanks for the rewording advice, will do that straight away

**Comment by magopian** [comment]
Ah, good catch on the link formats :) Changed the file to be a `.md`

**Comment by almet** [comment]
That looks good to me! You can land with or without my last nitpick !

**Pull request merged**